### ISSUE TITLE
Full Accessibility & 508 Office Audit Debt Resolution Team - Debt Letters V2

### ISSUE THREAD

**Issue by leahkeeler** [department-of-veterans-affairs/va.gov-team#22684]
## Steps to complete the Full Accessibility & 508 Audit: 
- [ ] VFS: Verify all accessibility defects that were identified at Staging Review have been fixed before requesting a full audit *(VSP understands that other teams may be accountable for fixing the defects identified at Staging Review. These defects are exempt and will not block a team from scheduling this touchpoint)*
- [ ] VFS Product Manager: Create this issue and fill in your team and feature name in the title
- [ ] VFS: Link to this issue once created in [#vfs-platform-support](https://dsva.slack.com/archives/CBU0KDSB1) in Slack; tag @ Trevor Pierce
- [ ] VSP: Create sub-issues attached to this epic with findings. Audits are performed on a first-come-first-serve basis and are scheduled with each biweekly sprint
- [ ] VFS: Coordinate with the VA 508 Office to address accessibility findings
- [ ] VSP: Update [Platform Collaboration Point Tracker](https://docs.google.com/spreadsheets/d/1OgPyEvUlNF6EnaYMFAXJkV6FKOvZnlPnbOQ2fAJ7W7A/edit#gid=1116695886)

## Artifacts
- staging/manage-va-debt
- Link to verification that accessibility checks are complete:
  - (preferred) Link to your [[TestRail VSP accessibility test plan](https://dsvavsp.testrail.io/index.php?/suites/view/8&group_by=cases:section_id&group_order=asc&display_deleted_cases=0&group_id=881) template](https://dsvavsp.testrail.io/index.php?/suites/view/14&group_by=cases:section_id&group_order=asc).
OR
  - Link to a doc in your product folder that indicates all required checks from the accessibility [staging review prep](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/staging-review-processes.md) doc are complete.
- List of known issues (with links to tickets when applicable)

## Additional information
As part of this review, VSP will send your product to VA's 508 office for approval (as applicable).

Please refer to the [Collaboration Cycle README](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/working-with-vsp/vsp-collaboration-cycle/README.md) on Github for more information about the Collaboration Cycle.

**Comment by AngelaFowler82** [comment]
Hello,
I am beginning the full accessibility audit for debt letters. Which test accounts should I use? User1 gives me a message saying "debt letters are not available at this time because something went wrong on our ent."

**Comment by joshkimux** [comment]
@leahkeeler have there been any updates with the test user accounts? 🙇

**Comment by noahgelman** [comment]
@leahkeeler Hi Leah, just following up on Josh's question.

**Comment by leahkeeler** [comment]
@noahgelman @joshkimux - My bad - I thought @craigwheeler answered in the slack channel - user42 works in staging - the actual debt letter downloads are dependent on vbms staging and is flaky -- I can give you the exact letter that is downloaded tho - it's a sample .pdf

**Comment by noahgelman** [comment]
@leahkeeler Then is this ready for a 508 audit? Or does it need more time?

**Comment by leahkeeler** [comment]
It's as ready as it will ever be - I has been live for quite some time -- the issue with vbms staging will eb persistent in the lower environments.

**Comment by AngelaFowler82** [comment]
Hello all,

Tried to download debt letters using User+42, and got this sad message.



 

Are the test accounts still not working?

**Comment by 1Copenut** [comment]
@leahkeeler Is there a preferred test user? The a11y specialists are having trouble with user 42, not sure if it's the user has no debt letter data or an API issue.

**Comment by leahkeeler** [comment]
@1Copenut @AngelaFowler82 - There is an issue with displaying debt letters right now that has to do with user rights - until it is resolved no one cna download. @craigwheeler will let you know when this is resolved. Sorry for the delay, this one is outside of our hands.

**Comment by joshkimux** [comment]
Any updates on this?

**Comment by craigwheeler** [comment]
Hi @joshkimux ... theres been some recent work around debt letters and appears that `vets.gov.user+1@gmail.com` is loading mocked user data for the download debt letter links now so you should be able to continue with user 1 let me know if something doesn't look right and i'll look into it ..thanks!

**Comment by AngelaFowler82** [comment]
FYI I am still not able to download debt letters using user+1.
![image](https://user-images.githubusercontent.com/78870095/125818391-f421f06c-6541-4b50-ba8b-f1ccf7dde987.png)

**Comment by craigwheeler** [comment]
@AngelaFowler82 @joshkimux sorry for the delay on this I believe the original issue has been resolved however turns out theres seems to be some intermittent issues with staging that are affecting users from hitting the debts endpoint which results in the error that were experiencing now... @kevinmirc and I are currently looking into the issue and we'll post back here once a fix has been confirmed.. again apologies for the back n forth but we're working on it and we'll post back soon!

**Comment by craigwheeler** [comment]
Hey @joshkimux and @AngelaFowler82 👋  ... Merged some changes on Thursday that seem to have fixed the issue so you should be able to access the download debt letters page on staging with `vets.gov.user+1@gmail.com`	now. Again if anything looks off lemme know thanks!

**Comment by joshkimux** [comment]
Thank you so much @craigwheeler ! We'll get started on this as soon as we have the availability to do so

**Comment by AngelaFowler82** [comment]
`Hate to break the bad news, but I am still unable to download debt letters. The first time I tried, after at least a minute of waiting I got a file which did not have an extention. The secont time I got the following: 
![image](https://user-images.githubusercontent.com/78870095/126192819-a9a57d7e-ea45-4078-9417-460a1c53736b.png)

I can tell you however that the download links do not specify the debt whose letter they are attached to, making it impossible for a screen reader user to determine which debt they are downloading a letter for.

**Comment by joshkimux** [comment]
<img width="653" alt="Screen Shot 2021-08-05 at 1 34 36 PM" src="https://user-images.githubusercontent.com/14154792/128395435-1e3f317c-b07b-4188-9f98-6e036227fb31.png">

**Comment by joshkimux** [comment]
<img width="412" alt="Screen Shot 2021-08-05 at 2 04 52 PM" src="https://user-images.githubusercontent.com/14154792/128399330-eeb8e762-2a92-42bc-9397-5ad8417c3cec.png">

This may need additional design QA @shiragoodman

**Comment by AngelaFowler82** [comment]
I am trying once again to review this component, and get a message stating there are no debt letters available on user+1. This review has been on our plates for over a month, and we would very much like to complete it. Could we please have a test account which can be thoroughly tested?

**Comment by joshkimux** [comment]
@1Copenut @noahgelman Our full 508 audit for debt letters is tentatively complete as @AngelaFowler82 was unable to access debt letters using user+1. 

@craigwheeler is there any chance you could work your magic to help us out there? 🙏

**Comment by leahkeeler** [comment]
Hey @1Copenut - Can we please push this into next sprint or the following? Here is the reason--> currently setting up a new e2e test environment that should be more robust - this is a vbms issue that requires a BE dev, we dont' have one - but hoping the new environment will solve this. Thank-you so much for your patience and understanding - we are barely keeping it above water with no BE dev right now.

**Comment by 1Copenut** [comment]
@shiragoodman Wanted to give you a heads up on [Leah's previous comment](https://github.com/department-of-veterans-affairs/va.gov-team/issues/22684#issuecomment-902056369). Josh and Angela have audited the test fails letter and are comfortable with their audit.

**Comment by leahkeeler** [comment]
Is this ticket good to close then?

**Comment by denisecoveyduc** [comment]
This epic seems to be linger for the past 3 years (wow!) not sure how it slipped thru the cracks so long but all work has been completed and may A11y Audits have been conducted since this work was done.

**Comment by shiragoodman** [comment]
thanks @denisecoveyduc ! We don't even have these full a11y audit ticket templates anymore... we stopped doing the pre-508 Office audit bc the a11y testing at Staging Review was sufficient enough before handing things over toe 508 Office. If you find anymore, feel free to close them!

**Comment by denisecoveyduc** [comment]
Thanks @shiragoodman  I was admittedly pretty surprised to find this one. There is always stuff hiding on the boards isn't there ;P  I will keep an eye out. Thank you!
